### PR TITLE
Fix grayscale passthrough

### PR DESCRIPTION
**Pull request by BytingBulldogs3539**

### CODE REVIEW

File: include/camera_model.h [hunk 1]
```diff
@@ -18,16 +18,18 @@
...
-};
-
+        Disconnected = 0,
+        OV5647, // Picam v1
+        IMX219, // Picam v2
+        IMX708, // Picam v3
+        IMX477, // Picam HQ
+        OV9281,
+        OV7251,
+        Unknown
```
Comment (mcm001) [inline include/camera_model.h:31]: Why did indentation change? You should run wpiformat

File: include/camera_model.h [hunk 2]
```diff
@@ -18,16 +18,18 @@
...
+    };
 CameraModel stringToModel(const std::string &model);
+bool isGrayScale(CameraModel model);
+static const CameraModel grayScaleCameras[] = {OV9281};
```
Comment (mcm001) [inline include/camera_model.h:35]: This should live in the cpp file

File: src/camera_grabber.cpp [hunk 3]
```diff
@@ -191,12 +191,10 @@ void CameraGrabber::setControls(libcamera::Request *request) {
...
                       m_settings.exposureTimeUs); // in microseconds
-        controls_.set(
-            libcamera::controls::FrameDurationLimits,
-            libcamera::Span<const int64_t, 2>{
-                {m_settings.exposureTimeUs,
-                 m_settings.exposureTimeUs}}); // Set default to zero, we have
-                                               // specified the exposure time
+        controls_.set(libcamera::controls::FrameDurationLimits,
+                      libcamera::Span<const int64_t, 2>{
+                          {1, 80000}}); // Set default to zero, we have
```
Comment (mcm001) [inline src/camera_grabber.cpp:196]: This comment seems wrong -- replace with why you chose these magic numbers?

File: src/camera_model.cpp [hunk 4]
```diff
@@ -39,3 +44,12 @@ CameraModel stringToModel(const std::string &model) {
     else
         return Unknown;
 }
+
+bool isGrayScale(CameraModel model) {
+    if (std::find(std::begin(grayScaleCameras), std::end(grayScaleCameras),
```
Comment (mcm001) [inline src/camera_model.cpp:49]: Should check if the found pointer is equal to the end of the vector.  Also, simplyify -- you don't need if (thing) return true, just return (thing)

File: src/libcamera_jni.cpp [hunk 5]
```diff
@@ -363,7 +363,7 @@ Java_org_photonvision_raspi_LibCameraJNI_awaitNewFrame
         return 0;
     }
 
-    MatPair *pair = new MatPair();
+    MatPair *pair = new MatPair(void);
```
Comment (mcm001) [inline src/libcamera_jni.cpp:366]: This will probably break in CI

**Review by mcm001 — CHANGES_REQUESTED** [review]

File: src/camera_grabber.cpp (hunk 3 above)
Comment (BytingBulldogs3539) [reply src/camera_grabber.cpp]: those should not have been committed...

File: src/gl_hsv_thresholder.cpp [hunk 6]
```diff
@@ -94,9 +95,10 @@ GLuint make_program(const char *vertex_source, const char *fragment_source) {
     return program;
 }
 
-GlHsvThresholder::GlHsvThresholder(int width, int height)
+GlHsvThresholder::GlHsvThresholder(int width, int height, CameraModel model)
     : m_width(width), m_height(height) {
     m_status = createHeadless();
+    useGrayScalePassThrough = isGrayScale(model);
```
Comment (mcm001) [inline src/gl_hsv_thresholder.cpp:101]: nit: initialize this as part of the top block. IE, `: m_width(width), m_height(height), useGrayScalePassThrough{isGrayScale(model)}`

**Review by mcm001 — APPROVED** [review]

**Pull request merged**